cpp/src/sampling/uniform_neighbor_sampling_impl.hpp

```diff
@@ -115,13 +115,165 @@ uniform_nbr_sample_impl(
 
     d_in = std::move(d_out_dst);
 
-    ++level;
+    ++hop;
   }
 
   return count_and_remove_duplicates<vertex_t, edge_t, weight_t>(
     handle, std::move(d_result_src), std::move(d_result_dst), std::move(*d_result_weights));
 #endif
 }
+
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          typename edge_type_t,
+          bool store_transposed,
+          bool multi_gpu>
+std::tuple<rmm::device_uvector<vertex_t>,
+           rmm::device_uvector<vertex_t>,
+           rmm::device_uvector<edge_t>,
+           std::optional<rmm::device_uvector<edge_type_t>>,
+           std::optional<rmm::device_uvector<weight_t>>,
+           rmm::device_uvector<int32_t>,
+           std::optional<rmm::device_uvector<int32_t>>>
+uniform_neighbor_sample_impl(
+  raft::handle_t const& handle,
+  graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu> const& graph_view,
+  std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+  std::optional<
+    edge_property_view_t<edge_t,
+                         thrust::zip_iterator<thrust::tuple<edge_t const*, edge_type_t const*>>>>
+    edge_type_view,
+  rmm::device_uvector<vertex_t>& d_in,
+  std::optional<rmm::device_uvector<int32_t>>& d_labels,
+  raft::host_span<const int> h_fan_out,
+  bool with_replacement,
+  uint64_t seed)
```

Comment:
Should we get a seed or better take a RngState?

Assuming this function will be called repeatedly, we will see the same sequence of generated random numbers for the same seed. Now, the randomness guarantee depends on providing different seeds in every function call (and this further complicates in multi-GPU cases).

Reply:
Changed to use `RngState`